clang/include/clang/Serialization/ASTReader.h

```diff
@@ -2527,7 +2527,7 @@ class BitsUnpacker {
 
 inline bool shouldSkipCheckingODR(const Decl *D) {
   return D->getASTContext().getLangOpts().SkipODRCheckInGMF &&
-         D->isFromGlobalModule();
+         (D->isFromGlobalModule() || !D->isInNamedModule());
```

Comment:
This may not be wanted. It will skip the ODR checks for the original clang modules. If we really want it, we should use `D->isFromHeaderUnits()`

Reply:
Added `isFromHeaderUnits()` and use it now.

But in general from my reading of C++ standard header units should treated as GMF see [10.3p5 note 2](https://eel.is/c++draft/module#import-note-2):
> [Note 2: A header unit is a separate translation unit with an independent set of defined macros. All declarations within a header unit are implicitly exported ([module.interface]), and are attached to the global module ([module.unit]). — end note]

So perhaps `isFromGlobalModule` should also include header units, it is used only in very few places. WDYT?

Reply:
They are different (and unfortunate) abstract layers. For example, according to the standard, everything not in a named module should be treated in the global module. However, in clang, in several places, we can't simply write `!isInNamedModule()` since we have clang modules and PCHs.

In clang we tried to be close to the wording of the spec but there are places (not limited to modules) they are not strictly the same.

Reply:
And it looks like this is not updated.

Reply:
Oops, forgot to push last commit :(
Please take another look.